chore(docs): updating the implementation of the theme switch button

yj-liuzepeng opened the pull request:
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

## Description

<img width="631" alt="image" src="https://github.com/element-plus/element-plus/assets/75007029/f1401ea1-395b-4685-bc25-d37120ef810e">


[docs/.vitepress/vitepress/components/common/vp-switch.vue](https://github.com/element-plus/element-plus/blob/dev/docs/.vitepress/vitepress/components/common/vp-switch.vue)

Now that the switch supports[ custom action icon](https://element-plus.gitee.io/en-US/component/switch.html#custom-action-icon), we can optimize the implementation of the theme switch button.

yj-liuzepeng committed:
chore(docs): updating the implementation of the theme switch button

btea commented:
Looks like there is a problem after the switch

![image](https://github.com/element-plus/element-plus/assets/24516654/1012f516-b32c-48ef-a35d-a6cfe87fbe44)

![image](https://github.com/element-plus/element-plus/assets/24516654/39fdcace-aaad-434d-838c-299109e85834)

yj-liuzepeng commented:
> Looks like there is a problem after the switch
> 
> ![image](https://user-images.githubusercontent.com/24516654/260602893-1012f516-b32c-48ef-a35d-a6cfe87fbe44.png)
> 
> ![image](https://user-images.githubusercontent.com/24516654/260602918-39fdcace-aaad-434d-838c-299109e85834.png)

I'm sorry 😬, it was working fine in the development environment. I'm not sure what went wrong. Let me take another look

btea committed:
test: [button] add comment resolve ts error (#13965)

* test: [button] add comment resolve ts error

* chore: update comment

makedopamine committed:
fix(components): [table-v2] calculate the width of MainTable in the same way (#13584)

* fix(components): [table-v2] calculate the width of MainTable in the same way

* fix: duplicate properties

FrontEndDog committed:
fix(components): [radio checkbox] fix click event triggered twice (#13936)

fix(components): [radio checkbox] @click event triggered twice

kamesan012 committed:
fix(components): fix the attribute conflict (#13504)

* fix(components): fix the attribute conflict

When value is input by shortcut, it's value will not be changed by defaultTime anymore

closed #13483

* fix(components): change param to pass the test

* fix(components): delete unnecessary changes

* fix(components): use let to replace ref variable

* fix(components): add test cases

kaine0923 commented on docs/.vitepress/vitepress/components/common/vp-theme-toggler.vue:
```diff
@@ -1,50 +1,25 @@
-    </ElIcon>
-  </VPSwitch>
+  <el-switch
+    :model-value="isDark"
```
cuz u use isDark which refs to the native browser API so u should use ClientOnly to wrap it

yj-liuzepeng committed:
chore(docs): updating the implementation of the theme switch button

yj-liuzepeng committed:
Merge branch 'update/theme-switch' of https://github.com/yj-liuzepeng/element-plus into update/theme-switch

yj-liuzepeng committed:
Update
